### ISSUE TITLE
DigitalSampleState is always None

### ISSUE THREAD

**Issue by PaulNoto** [jefffhaynes/XBee#12]
This is my main issue at this time. Samples are coming in from my router and End Devices. However, the DigitalSampleState is always None. I can send commands to my router to control output pins ( It's a Flood Light) and the light turns on and off. It also has a motion detector when the Motion detector is triggered the Controller gets a Sample. However the DigitalSampleState is None. Because I have a new S2C XBee installed in the Flood Light I wanted to make sure it worked in my active network. So, I moved the Router from my test network to my active network ( where it came from originally ) by changing it's PAN ID and API Mode from 1 to 2. It works fine. The GBee Library on the Netduino picks up Pin State and reports motion. I moved it back to my test network, and Downloaded the source code to your BinarySerializer Library, so I could step through it with the Debugger and try to figure out why the DigitalSampleState wasn't getting decoded.

WOW ! What an elegant piece of software! Way beyond my skill level. I bet you graduated top of your class from MIT.

Anyway I attempted to keep up with what it was doing. However, at the part where it checks the DigitalSampleState, I could not figure out why it was deciding that there was None.

I'm stumped. Any help would be greatly appreciated.

**Comment by jefffhaynes** [comment]
Thanks, much appreciated!  I'm going to try to work on this tonight. 






On Sun, Apr 24, 2016 at 3:12 PM -0700, "PaulNoto" <notifications@github.com> wrote:












This is my main issue at this time. Samples are coming in from my router and End Devices. However, the DigitalSampleState is always None. I can send commands to my router to control output pins ( It's a Flood Light) and the light turns on and off. It also has a motion detector when the Motion detector is triggered the Controller gets a Sample. However the DigitalSampleState is None. Because I have a new S2C XBee installed in the Flood Light I wanted to make sure it worked in my active network. So, I moved the Router from my test network to my active network ( where it came from originally ) by changing it's PAN ID and API Mode from 1 to 2. It works fine. The GBee Library on the Netduino picks up Pin State and reports motion. I moved it back to my test network, and Downloaded the source code to your BinarySerializer Library, so I could step through it with the Debugger and try to figure out why the DigitalSampleState wasn't getting decoded.



WOW ! What an elegant piece of software! Way beyond my skill level. I bet you graduated top of your class from MIT.



Anyway I attempted to keep up with what it was doing. However, at the part where it checks the DigitalSampleState, I could not figure out why it was deciding that there was None.



I'm stumped. Any help would be greatly appreciated.
